code/segmentedtickerplant/datastripe.q

```diff
@@ -1,60 +1,64 @@
+.ds.stripeconfig:@[value;`.ds.stripeconfig;`striping.json];
 
-.ds.segmentconfig:@[value;`.ds.segmentconfig;`segmenting.csv];
-.ds.filtermap:@[value;`.ds.filtermap;`filtermap.csv];
-
-//if statement checks segmenting.csv and filtermap.csv exist. If not, process exited and message sent to error logs.
-//configload transfoms segmenting.csv into table format so it can be accessed.
-//configload transforms filtermap.csv into table format then into a mapping of wcRef to filter which can be accessed and applied to data.
-//checks csv files load correctly. If not, process exited and message sent to error logs.
+//Loads the striping.json config file checks if each subscriptiondefault is set for each segment and errors if not defined
 
 configload:{
-     scpath:first .proc.getconfigfile[string .ds.segmentconfig];
-     fmpath:first .proc.getconfigfile[string .ds.filtermap];
-     {if[()~key hsym x;.lg.e[`init;"The following file can not be found: ",string x]]} each (scpath;fmpath);
-     @[{.stpps.segmentconfig:("SIS";enlist",")0: hsym x};scpath;{.lg.e[`init;"Failure in loading ",string y]}[;scpath]];
-     @[{.stpps.segmentfiltermap:(!/)(("S*";enlist",")0: hsym x)`wcRef`filter};fmpath;{.lg.e[`init;"Failure in loading ",string y]}[;fmpath]];
+     scpath:first .proc.getconfigfile[string .ds.stripeconfig];
+     {if[()~key hsym x;.lg.e[`init;"The following file can not be found: ",string scpath]]};
+     .stpps.stripeconfig:.j.k raze read0 scpath;
+     defaults:{first (flip .stpps.stripeconfig[x])[`subscriptiondefault]}each key .stpps.stripeconfig;
+     errors:1+ where {[x] not ("ignore"~x) or ("all"~x)}each defaults;
+     {if[0<count errors;.lg.e[`sub;m:"subscriptiondefault not defined for segment/segments ",string[x]," "]]}each errors;
      };
 
 initdatastripe:{
      .lg.o[`init;"init datastriping"];
      configload[];
      };
 
+
 \d .stpps
 
-// Function to map the where clause from config table extracted by .stpps.segmentfilter function to tablename
-// Allows use of ` as argument for tables
+//makes a dictionary of tables and there filters for segmentedsubdetails
+
 filtermap:{[tabs;id] if[tabs~`;tabs:.stpps.t]; ((),tabs)!.stpps.segmentfilter\:[(),tabs;id]}
 
-// Find where clause from config tables
+//grabs filters from the config files and for the "ignoretable" filter converts to "" to allow segmentedsudetails to run
+
 segmentfilter:{[tbl;segid]
-     wcref:first exec wcRef from .stpps.segmentconfig where table=tbl , segmentID=segid;
-     .stpps.segmentfiltermap[wcref]
+     id:`$string segid;
+     filter:first (flip .stpps.stripeconfig[id])[tbl];
+     $[filter~"ignoretable";filter:"";filter]
      };
 
-// Subscribe to particular segment using segmentID based on .u.sub
+//subscribe to a table using segmentID to determine filtering
 subsegment:{[tbl;segid];
-     //tablename and segmentid used to get filters
+//casting segid to an symbol as json is restrictive
+     id:`$string segid;
```

Comment:
I think we're doing this because the keys in a dict parsed from a JSON object will always be symbols, right? Should we just have it that segment IDs _are_ symbols, rather than numbers? i.e. in this function, expect a symbol to be passed rather than a number

Potentially this allows a bit more flexibility in giving segments meaningful names. e.g. if you had a segment that filtered all tables to ``exch=`LSE``, you could name that segment `` `LSE`` rather than having to give it a number

Just a thought, not sure if there are other places in the code were changes would have to be made to accomodate a symbol based segment name